### PR TITLE
Consolidate Link props

### PR DESCRIPTION
**Pull request by carkom**
Consolidates AnchorLink, ButtonLink and LinkButton to use existing props (Link and Button - with type).

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -146,10 +146,10 @@ module.exports = {
           'Using default React import is discouraged, please use named exports directly instead.',
       },
       {
-        // forbid <a> in favor of <LinkButton> or <ExternalLink>
+        // forbid <a> in favor of <Button type={'link'}> or <ExternalLink>
         selector: 'JSXOpeningElement[name.name="a"]',
         message:
-          'Using <a> is discouraged, please use <LinkButton> or <ExternalLink> instead.',
+          'Using <a> is discouraged, please use <Button type={link}> or <ExternalLink> instead.',
```
Comment (MatissJanis) [inline .eslintrc.js:152]: **💬 suggestion:** ‏
```suggestion
          'Using <a> is discouraged, please use <Button type="link"> or <ExternalLink> instead.',
```

File: .eslintrc.js [hunk 2]
```diff
@@ -146,10 +146,10 @@ module.exports = {
           'Using default React import is discouraged, please use named exports directly instead.',
       },
       {
-        // forbid <a> in favor of <LinkButton> or <ExternalLink>
+        // forbid <a> in favor of <Button type={'link'}> or <ExternalLink>
```
Comment (MatissJanis) [inline .eslintrc.js:149]: **💬 suggestion:** ‏
```suggestion
        // forbid <a> in favor of <Button type="link"> or <ExternalLink>
```

**Review by MatissJanis — APPROVED** [review]
This is definitely an improvement!

What still confuses me though is: `<Link type="button">` and `<Button type="link">`. We should eventually get rid of one of these.

**Comment by carkom** [comment]
> This is definitely an improvement!
> 
> What still confuses me though is: `<Link type="button">` and `<Button type="link">`. We should eventually get rid of one of these.

<Link type="button"> - is an element that looks like a button but uses a "to" prop for navigating inside the app (eg. "Back" buttons).

`<Button type="link">` - does not really need to be a button because it doesn't actually look like a button. It looks like a link but uses an "onClick" prop to call a function rather than navigate to another page.

All the other "link" types we use need a "to" prop sent to them in order to not error out. I guess the second option above is a work around for making something look like a link but function like a button.

**Comment by carkom** [comment]
I've created a new "ClickLink" in the Link file to be used instead of the `<Button type="link">`. 

Thoughts?

**Comment by carkom** [comment]
@MatissJanis - all our link options are under one roof. I moved over external links as well as removing the "link" type in `<Button>`. I renamed all the options so they are more logical and easier to follow.

**Review by MatissJanis — APPROVED** [review]
This is great!

**Comment by MatissJanis** [comment]
Closes #1414

**Pull request merged**